packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -827,6 +841,27 @@ class ThemeData extends Diagnosticable {
   }
 }
 
+/// Describes whether descendent Material Design widgets of various types
+/// should automatically adapt to use Cupertino widgets when running on iOS.
+@immutable
+class AdaptiveWidgetThemeData {
+  const AdaptiveWidgetThemeData(this._adaptivenessOptions);
+
+  static const AdaptiveWidgetThemeData none =
+      const AdaptiveWidgetThemeData(const <Type, bool>{
+        Switch: false,
+      });
+
+  static const AdaptiveWidgetThemeData all =
```

Comment:
this map can't be complete, since we don't know what other widgets the user may have invented.

Reply:
i would treat "null" as meaning "yes always".

Reply:
or have a `AdaptiveWidgetThemeData.all` constructor which returns a subclass that always returns true

Reply:
I was gonna use null as a signal to deal with merging. Otherwise, a child with limited knowledge might end up always overwriting everything. Maybe there are other ways.

Reply:
But ya, you're right that it can't be called 'all'. Maybe more like allBundledWidgets or something on which users should call ..addAll(their own stuff)